### ISSUE TITLE
HEEX template formatting not working - formatter plugin issue

### ISSUE THREAD

**Issue by Benjamin-van-Heerden** [elixir-lsp/elixir-ls#1152]
### Environment

* Elixir & Erlang versions (elixir --version):  1.18.1-otp-27
* Elixir Language Server version: 0.26.0
* Operating system:  Linux
* Editor or IDE name (e.g. Emacs/VSCode): VSCode and Zed (all indication is that it will be an issue everywhere) 
* Editor Plugin/LSP Client name and version: ElixirLS normal VSCode plugin

### Current behavior

When trying to format a .ex or .heex file I get a "Unable to get formatter options..." and formatting does not work

.formatter.exs:
```
[
  import_deps: [:ecto, :ecto_sql, :phoenix],
  subdirectories: ["priv/*/migrations"],
  plugins: [Phoenix.LiveView.HTMLFormatter],
  inputs: ["*.{heex,ex,exs}", "{config,lib,test}/**/*.{heex,ex,exs}", "priv/*/seeds.exs"]
]
```

When I comment out the plugins, formatting works again for .ex files, but not for .heex. HEEX templates are an essential part of my development workflow so this is really frustrating.

Logs:
```
Unable to get formatter options for /home/benjamin/Documents/studii/code/studii-main/lib/studii_main_web/live/about/index.ex: ** (Protocol.UndefinedError) protocol Enumerable not implemented for type Atom

Got value:

    :ok

    (elixir 1.18.1) lib/enum.ex:1: Enumerable.impl_for!/1
    (elixir 1.18.1) lib/enum.ex:166: Enumerable.reduce/3
    (elixir 1.18.1) lib/enum.ex:4511: Enum.reduce/3
    (language_server 0.26.0) lib/language_server/mix_tasks/format.ex:339: Mix.Tasks.ElixirLSFormat.load_plugins/2
    (language_server 0.26.0) lib/language_server/mix_tasks/format.ex:436: Mix.Tasks.ElixirLSFormat.formatter_for_file/2
    (language_server 0.26.0) lib/language_server/source_file.ex:282: ElixirLS.LanguageServer.SourceFile.formatter_for/3
    (language_server 0.26.0) lib/language_server/server.ex:1098: ElixirLS.LanguageServer.Server.handle_request/2
    (language_server 0.26.0) lib/language_server/server.ex:843: ElixirLS.LanguageServer.Server.handle_request_packet/3
    (language_server 0.26.0) lib/language_server/server.ex:302: ElixirLS.LanguageServer.Server.handle_cast/2
    (stdlib 4.3.1.6) gen_server.erl:1123: :gen_server.try_dispatch/4
    (stdlib 4.3.1.6) gen_server.erl:1200: :gen_server.handle_msg/6
    (stdlib 4.3.1.6) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

---
## Cause of the issue
After some digging, I am convinced that commit `a1269e0` contains the problem.
Specifically, in https://github.com/elixir-lsp/elixir-ls/blob/master/apps/language_server/lib/language_server/source_file.ex around line 260-280, the following code for the "plugin_loader" just returning :ok

```
 if mix_project? do
        if MixProjectCache.loaded?() do
          opts = [
            deps_paths: MixProjectCache.deps_paths(),
            manifest_path: MixProjectCache.manifest_path(),
            config_mtime: MixProjectCache.config_mtime(),
            mix_project: MixProjectCache.get(),
            root: project_dir,
            plugin_loader: fn _plugins ->
              # we don't do any plugin loading as this may trigger compile
              # TODO it may be safe to compile on 1.18+
              :ok
            end
          ]
```

Obviously the TODO is there for a reason, maybe some clarity on this would be much appreciated.

I think that this will also break formatting on every Phoenix project using 0.26.0 - don't know if it goes further back than this.

Maybe hard coded fix for Phoenix.LiveView.HTMLFormatter to allow that one through?

**Comment by lukaszsamson** [comment]
Fixed in v0.26.1

**Comment by Benjamin-van-Heerden** [comment]
Dude, you're a legend!

Literally, while I finished typing up the issue the CI pipeline triggered

Thanks
